tests/tests.cpp

```diff
@@ -1522,6 +1523,312 @@ TEST_F(EngineBasicTest, TestSortedCustomCompareFunction) {
   delete engine;
 }
 
+// ========================= Sync Point ======================================
+
+#if DEBUG_LEVEL > 0
+TEST_F(EngineBasicTest, TestBatchWriteSyncPoint) {
+  // SyncPoint
+  {
+    Configs test_config = configs;
+    test_config.max_write_threads = 16;
+    ASSERT_EQ(Engine::Open(db_path.c_str(), &engine, test_config, stdout),
+              Status::Ok);
+
+    int cnt = 20;
+    std::atomic<bool> write_batch{false};
+
+    SyncPoint::GetInstance()->LoadDependency(
+        {{"Test::Set::Finish::1",
+          "KVEngine::BatchWrite::AllocateRecord::After"}});
+    SyncPoint::GetInstance()->SetCallBack(
+        "KVEngine::BatchWrite::AllocateRecord::After", [&](void *size) {
+          size_t bsize = *((size_t *)size);
+          ASSERT_EQ(cnt, bsize);
+        });
+    SyncPoint::GetInstance()->SetCallBack(
+        "KVEngine::BatchWrite::Pesistent::0",
+        [&](void *) { write_batch.store(true); });
+
+    SyncPoint::GetInstance()->SetCallBack("Test::Set::Start::0", [&](void *) {
+      while (!write_batch.load())
+        ;
+    });
+
+    SyncPoint::GetInstance()->SetCallBack("Test::Delete::Start::0",
+                                          [&](void *) {
+                                            while (!write_batch.load())
+                                              ;
+                                          });
+
+    SyncPoint::GetInstance()->EnableProcessing();
+
+    std::vector<std::thread> ts;
+
+    ts.emplace_back([&]() {
+      WriteBatch wb;
+      for (int i = 0; i < cnt; ++i) {
+        std::string key = "key" + std::to_string(i);
+        std::string val = "val" + std::to_string(i);
+        wb.Put(key, val);
+      }
+      ASSERT_EQ(engine->BatchWrite(wb), Status::Ok);
+    });
+
+    ts.emplace_back([&]() {
+      TEST_SYNC_POINT("Test::Set::Start::0");
```

Comment:
Please give these Sync Points a meaningful name.
like "Test::before-set-k10v17", "Test::after-set-k8-v15". This can improve readability.